core/lib/dal/src/events_web3_dal.rs

```diff
@@ -97,21 +106,56 @@ impl EventsWeb3Dal<'_, '_> {
                 let topics: Vec<_> = topics.iter().map(H256::as_bytes).collect();
                 query = query.bind(topics);
             }
+
             query = query.bind(limit as i32);
 
-            let db_logs: Vec<StorageWeb3Log> = query
+            let mut db_logs: Vec<StorageWeb3Log> = query
                 .instrument("get_logs")
                 .report_latency()
                 .with_arg("filter", &filter)
                 .with_arg("limit", &limit)
                 .fetch_all(self.storage.conn())
                 .await?;
+
+            if api_mode == APIMode::Modern {
+                db_logs.retain(|log| {
+                    log.address != L2_ETH_TOKEN_ADDRESS.as_bytes()
+                        || log.topic1 != TRANSFER_EVENT_TOPIC.as_bytes()
+                });
```

Comment:
Dumb question: isn't such events filtered out on the DB query level?

Reply:
Right here they are not. The thing is, that we need to somehow deal with indexes, so there will be no gaps if we get only filtered events. Generally we can put this filtering inside query, but it seems a bit more painful than doing it this way